jcl/src/java.base/share/classes/java/lang/StringBuffer.java

```diff
@@ -877,7 +887,7 @@ public synchronized StringBuffer insert(int index, char[] chars) {
 			} else {
 				// Check if the StringBuffer is compressed
 				if (count >= 0) {
-					decompress(value.length);
+					decompress(value.length, currentLength + chars.length);
```

Comment:
The first argument should specify the minimum required capacity; it should also be `currentLength + chars.length` - a local capturing that value seems appropriate. Same goes for all uses in `insert()` methods.